src/hotspot/share/memory/dynamicArchive.cpp

```diff
@@ -367,6 +368,15 @@ void DynamicArchive::dump() {
     return;
   }
 
+  // regenerate lambdaform holder classes
+  if (LambdaFormInvokers::should_regenerate_holder_classes()) {
+    JavaThread* THREAD = JavaThread::current();
+    assert(THREAD->is_Java_thread(), "Should be JavaThread");
+    log_info(cds, dynamic)("Regenerate lambdaform holder classes ...");
+    LambdaFormInvokers::regenerate_holder_classes(THREAD);
```

Comment:
What if this throws an exception?

Reply:
If there is exception in java side to regenerate the holder classes, there is a log for cds to indicate the regeneration failed, the exception is cleared and continue. It will not affect other parts.